src/lightcurvelynx/models/sed_curve_model.py

```diff
@@ -0,0 +1,520 @@
+        self.interp = RectBivariateSpline(
+            self.times,
+            self.wavelengths,
+            sed_values,
+            kx=interp_degree,
+            ky=interp_degree,
+        )
+
+    def evaluate_sed(self, times, wavelengths):
+        """Evaluate the SED at the given times and wavelengths.
+
+        Parameters
+        ----------
+        times : np.ndarray
+            A length T array of times (in the given units) at which to evaluate the SED
+            (relative to t0).
+        wavelengths : np.ndarray
+            A length W array of wavelengths (in the given units) at which to evaluate the SED.
+
+        Returns
+        -------
+        sed_values : np.ndarray
+            A (T x W) matrix of SED values (in the given units) at the given times and wavelengths.
+        """
+        if self.period is None:
+            sed_values = np.zeros((len(times), len(wavelengths)))
+
+            in_range = (times >= self.times[0]) & (times <= self.times[-1])
+            sed_values[in_range, :] = self.interp(times[in_range], wavelengths, grid=True)
+
+            if self.baseline is not None:
+                sed_values[~in_range, :] = self.baseline[np.newaxis, :]
+        else:
+            # Create the modulo times for periodic evaluation and an inverse mapping to original order.
+            times = np.mod(times, self.period)
+            argsort_idx = np.argsort(times)
+            inv_idx = np.zeros_like(argsort_idx)
+            inv_idx[argsort_idx] = np.arange(len(times))
+
+            sed_values = self.interp(times[argsort_idx], wavelengths, grid=True)
+            sed_values = sed_values[inv_idx, :]
+        return sed_values
+
+    @staticmethod
+    def _three_column_to_matrix(data):
+        """Convert 3-column SED data to a matrix form.
+
+        Parameters
+        ----------
+        data : np.ndarray
+            A 2D array of shape (N x 3) containing phases, wavelengths, and fluxes.
+
+        Returns
+        -------
+        unique_phases : np.ndarray
+            A length T array of unique phases.
+        unique_wavelengths : np.ndarray
+            A length W array of unique wavelengths.
+        lightcurve_matrix : np.ndarray
+            A 2D array of shape (T x W) with fluxes.
+        """
+        phases = data[:, 0]
+        wavelengths = data[:, 1]
+        fluxes = data[:, 2]
+
+        unique_wavelengths = np.unique(wavelengths)
+        unique_phases = np.unique(phases)
+
+        lightcurve_matrix = np.zeros((len(unique_phases), len(unique_wavelengths)))
+        for wave, phase, flux_val in zip(wavelengths, phases, fluxes, strict=False):
+            wave_idx = np.where(unique_wavelengths == wave)[0][0]
+            phase_idx = np.where(unique_phases == phase)[0][0]
+            lightcurve_matrix[phase_idx, wave_idx] = flux_val
+
+        return unique_phases, unique_wavelengths, lightcurve_matrix
+
+
+class SEDCurveModel(SEDModel):
```

Comment:
Again, let's try to avoid lightcurves in the context of SED models (until we talk about generated bandfluxes)